Source/WebKit/Platform/IPC/Connection.h

```diff
@@ -584,7 +589,7 @@ uint64_t Connection::sendWithAsyncReply(T&& message, C&& completionHandler, uint
             T::callReply(*decoder, WTFMove(completionHandler));
         else
             T::cancelReply(WTFMove(completionHandler));
-    }, CompletionHandlerCallThread::MainThread));
```

Comment:
I think this is wrong, because we're changing MainThread to (implicitly) the current thread.

I think the contract is that this reply handler will execute on the connection runloop, whereas the current thread can be anything.

I don't think there's an easy way to express that currently, so this probably needs to be `AnyThread` (though we could use MainThread if that's what the connection runloop is?).

Reply:
Yeah, this is covered in one of the tests as a comment and disabled test.
We can construct a threadlike assertion out of the target runloop, but as the whole api contract is problematic, it's better to just not do that.
Later on the api contract should be changed to contract where the async handler is invoked in the calling runloop. This api contract should make more sense.